src/assets/scripts/client/navigationLibrary/StandardRoute/StandardRouteWaypointModel.js

```diff
@@ -283,23 +283,31 @@ export default class StandardRouteWaypointModel extends BaseModel {
      * @for StandardRouteWaypointModel
      * @method clonePositionFromFix
      * @param fixCollection {FixCollection}
+     * @return {StandardRouteWaypointModel}
      * @private
+     * @chainable
      */
     clonePositionFromFix() {
-        if (this.name[0] === VECTOR_WAYPOINT_PREFIX) {
+        const isFlyOverWaypoint = this.name.indexOf(FLY_OVER_WAYPOINT_PREFIX) !== -1;
+        const isHoldWaypoint = this.name.indexOf(HOLD_WAYPOINT_PREFIX) !== -1;
+        const isVectorWaypoint = this.name.indexOf(VECTOR_WAYPOINT_PREFIX) !== -1;
```

Comment:
you're not thinking 4th dimensionally.  I'm talking about the `RouteModel` (which should actually be called the `RouteStringModel`.  We have static methods on that class to help in these types of circumstances:

```javascript
RouteModel.isVectorRouteString(this.name)
```